### PR TITLE
Ieee802154/radio driver api extension

### PR DESCRIPTION
**Pull request by rlubos**

### CODE REVIEW

File: include/net/ieee802154_radio.h [hunk 1]
```diff
@@ -54,14 +54,44 @@ struct ieee802154_filter {
 /* @endcond */
 };
 
+/** IEEE802.15.4 Transmission mode. */
+enum ieee802154_tx_mode {
+	/** Transmit packet immediately, no CCA. */
+	IEEE802154_TX_MODE_PLAIN,
```
Comment (wbober) [inline include/net/ieee802154_radio.h:61]: ```IEEE802154_TX_MODE_NO_CCA``` to be more explicit?

File: include/net/ieee802154_radio.h [hunk 2]
```diff
@@ -109,6 +121,7 @@ struct ieee802154_config {
 		/** ``IEEE802154_CONFIG_AUTO_ACK_FPB`` */
 		struct {
 			bool enabled;
+			enum ieee802154_fpb_mode mode;
```
Comment (rlubos) [inline include/net/ieee802154_radio.h]: Now I wonder, if we should just add `IEEE802154_FPB_ADDR_MATCH_DISABLED` and get rid of the bool.

File: include/net/ieee802154_radio.h (hunk 1 above)
Comment (rlubos) [reply include/net/ieee802154_radio.h:61]: I wanted to make `_NO_CCA` implicit, so I don't need to add it to the delayed variant as well. But  I can see that `_PLAIN` might not sound too good in terms of transmission. How about `IEEE802154_TX_MODE_DIRECT`/`IEEE802154_TX_MODE_IMMEDIATE`/`IEEE802154_TX_MODE_INSTANT`? Any other ideas?

File: include/net/ieee802154_radio.h [hunk 3]
```diff
@@ -54,14 +54,44 @@ struct ieee802154_filter {
...
+	IEEE802154_TX_MODE_PLAIN,
+
+	/** Perform CCA before packet trasmission. */
+	IEEE802154_TX_MODE_CCA,
+
+	/** Perform full CSMA CA procedure before packet transmission. */
+	IEEE802154_TX_MODE_CSMA_CA,
+
+	/** Transmit packet in the future, at specified time, no CCA. */
+	IEEE802154_TX_MODE_TXTIME,
```
Comment (MarekPorwisz) [inline include/net/ieee802154_radio.h:70]: don't we need additional capability for delaying tx?

Comment (rlubos) [reply include/net/ieee802154_radio.h:70]: Good point, will add.

**Pull request closed without merging**